### ISSUE TITLE
Slow performance when navigating local folders

### ISSUE THREAD

**Issue by fulalas** [linuxmint/nemo#3385]
### Distribution

PorteuX

### Package version

6.0.2

### Frequency

Always

### Bug description

In comparison to dolphin (KDE), pcmanfm-qt (LXQt) and thunar (Xfce), nemo takes much more time to navigate local folders, reaching 500ms in a live system where the filesystem is 100% in RAM.

### Steps to reproduce

1. Open folder A
2. Double click on subfolder B

### Expected behavior

The subfolder content should be rendered in such a way that we shouldn't notice any delay.

### Additional information

This is not distro related (also tested on Mint 21.3).

This is not hardware related (tested in a physical machine Ryzen 7 7840HS + 32 GB + nvme).

**Comment by Crispycjr** [comment]
I really want more attention on this.

I remember Ubuntu Budgie used to have this set up to where it wasn't an issue, that was a few years ago.

**Comment by Jeremy7701** [comment]
You don't say how many files/folders are in sub-folder B.
Also, you don't say what file system is used.
You mention a "...in a live system where the filesystem is 100% in RAM"
A live system is generally taken to mean one booted from DVD/USB and then it runs everything in memory - of course everything runs **much slower** than booting a properly installed system. It's only normally used to get a "feel" for a distro.

**Comment by fulalas** [comment]
It doesn't matter the filesystem, including but not restricted to overlay/aufs. It's not an IO bottleneck; it's an UI bottleneck. It happens consistently, in any scenario, regardless of the filesystem or the amount of files/folders.

There's a noticeable delay when navigating from folder A to B regardless. You can feel **and** measure that. And this is something introduced kind of recently.

As I said, you can compare to other file managers, like dolphin (KDE), pcmanfm-qt (LXQt) and thunar (Xfce), but not Nautilus, of course.

**Comment by Crispycjr** [comment]
This has been around a bit longer than recently, at least 3+ years in my experiences using different distributions that ship with nemo. Even nautilus has a considerable speed advantage in switching directories, it's difficult to ignore once you recognize it.

Nemo is a great file manager, one of the best in the linux environment. This is just one of those polish jobs that would make it noticably smoother and finalize it as comparable to Dolphin. I'm not sure how hard it would be, though.

**Comment by Jeremy7701** [comment]
Just tried it on nemo in LMDE6
Hard to estimate speed, since I'm only human, but icon-style gets displayed, with thumbnails in maybe 100msec?
Not quite instant, but certainly not half a second.
Unlikely that data is already in RAM, but it might be.
There is an additional (just perceptible) delay before the count of displayed items and display of free space in the 'disk' file system on the Sata SSD appears.

**Comment by nioncode** [comment]
I'm also experiencing extremely slow listing of folders (I don't have a reference with older nemo versions, since I just recently started using nemo). When opening one of my directories, it consistently takes 7 seconds to list any file / directory when starting with a fresh instance of nemo. After entering the directory again, files + folders get displayed instantly and only the count of files in each subdir refreshes asynchronously. This delay goes away if I disable counting of files in the preferences, so it seems like something is blocking here (interestingly afterwards the count of files in the subfolder refreshes asynchronously, so I guess it is supposed to happen asynchronously, but something is still blocking the main listing from appearing?).

There seem to be many other issues with the performance:
* when entering a directory, sometimes folders (and some files) are visible immediately and then other files start to pop up 1-2s later (this goes away when disabling thumbnails / file previews in the settings)
* general ui 'slugginesh' if I just navigate back and forth between a folder via `alt+left` and `alt+right`, sometimes the path does not even update correctly if you don't wait for the folder to load completely
* in general, loading a folder is way slower than on other file managers (e.g. I used `spacefm` in the past and entering folders there is more or less instant. when switching to nemo you think you are on a hard drive and not on an ssd). Even after entering a directory once, you'd assume it is now in the disk cache of the OS, so listing files again should be instant; but in nemo it takes multiple hundred milliseconds, which makes it more likely that it has to do something with the ui than the raw file listing performance

**Comment by fulalas** [comment]
It seems this has been fixed: https://blog.linuxmint.com/?p=5025

**Comment by nioncode** [comment]
Indeed, I retested with https://github.com/linuxmint/nemo/commit/80544ae2c7c8c4c2165aaac00e5031614537be86 and my 'problematic' folders are loaded now way faster and appear almost instantly. This can probably be closed then.
